src/app/_components/ClientPagination.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import clsx from 'clsx'
+
+type PaginationProps = {
+  currentPage: number
+  total: number
+  size: number
+  setCurrentPage: (page: number) => void
+}
+
+export function ClientPagination({
+  currentPage,
+  setCurrentPage,
+  total,
+  size,
+}: PaginationProps) {
+  const pages = Math.ceil(total / size)
+
+  return (
+    <div className="flex justify-between gap-4 rounded-lg bg-brand-300 p-2 text-brand-700">
+      {/* Prev */}
+      <button
+        className="rounded px-3 py-1 transition hover:bg-brand-400/30"
+        onClick={() => {
+          if (currentPage > 1) setCurrentPage(currentPage - 1)
+        }}
+      >
```

Comment:
I suggest extracting this logic

```
  function handlePrevClick() {
    if (currentPage > 1) {
      setCurrentPage(currentPage - 1)
    }
  }

  ```

```suggestion
        onClick={handlePrevClick}
      >
```

Reply:
Or `onClick={() => currentPage > 1 && setCurrentPage(currentPage - 1)}`